### PR TITLE
Migrate deprecated fields

### LINKED ISSUE #3996: Migrate from `deprecated` fields
Some GraphQL fields were recently deprecated in our API:
- OrderLine.thumbnailUrl
- Page.availableOn
- Page.isVisible
- ProductVariant.availability
- Product.availability -> pricing
- Product.availableOn
- Product.price
- Product.thumbnailUrl
- Collection.publishedDate

We should migrate queries in Dashboard 2.0 to use the new fields, so that we'll be able to drop the old ones from the schema.

### PR DESCRIPTION
**Pull request by benekex2**
I want to merge this change because...

Resolves #3996

### Pull Request Checklist

1. [x] Privileged views and APIs are guarded by proper permission checks.
1. [x] All visible strings are translated with proper context.
1. [x] All data-formatting is locale-aware (dates, numbers, and so on).
1. [x] Database queries are optimized and the number of queries is constant.
1. [x] Database migration files are up to date.
1. [ ] The changes are tested.
1. [ ] The code is documented (docstrings, project documentation).
1. [x] GraphQL schema and type definitions are up to date.
1. [x] Changes are mentioned in the changelog.

### CODE REVIEW

File: saleor/static/dashboard-next/categories/queries.ts [hunk 1]
```diff
@@ -100,16 +100,16 @@ export const categoryDetails = gql`
           node {
             id
             name
-            availability {
+            basePrice {
+              amount
+              currency
+            }
+            pricing {
```
Comment (maarcingebala) [inline saleor/static/dashboard-next/categories/queries.ts]: `pricing.available` is there for compatibility but it's also deprecated. We should use `product.isAvailable` for that purpose.

File: saleor/static/dashboard-next/products/queries.ts [hunk 2]
```diff
@@ -52,10 +52,10 @@ export const productFragment = gql`
     thumbnail {
       url
     }
-    availability {
+    pricing {
```
Comment (maarcingebala) [inline saleor/static/dashboard-next/products/queries.ts]: Here as well we should instead use `product.isAvailable`.

File: saleor/static/dashboard-next/products/queries.ts [hunk 3]
```diff
@@ -115,7 +115,7 @@ export const productFragmentDetails = gql`
         slug
       }
     }
-    availability {
+    pricing {
```
Comment (maarcingebala) [inline saleor/static/dashboard-next/products/queries.ts]: Below we also use `pricing.available` while we should use `product.isAvailable`.

**Comment by maarcingebala** [comment]
@benekex2 Please check if we got rid of all `Product.pricing.available` usages, I forgot to mention that in the issue.

File: saleor/static/dashboard-next/categories/components/CategoryProducts/CategoryProducts.tsx [hunk 4]
```diff
@@ -95,7 +97,7 @@ export const ProductList = withStyles(styles, { name: "ProductList" })(
             products,
             product => (
               <TableRow key={product ? product.id : "skeleton"}>
-                <TableCellAvatar thumbnail={product && product.thumbnailUrl} />
+                <TableCellAvatar thumbnail={product && product.thumbnail.url} />
```
Comment (dominik-zeglen) [inline saleor/static/dashboard-next/categories/components/CategoryProducts/CategoryProducts.tsx]: We should use `maybe(() => product.thumbnail.url)` here

File: saleor/static/dashboard-next/components/ProductList/ProductList.tsx [hunk 5]
```diff
@@ -154,18 +154,18 @@ export const ProductList = withStyles(styles, { name: "ProductList" })(
                 </TableCell>
                 <TableCell className={classes.colPublished}>
                   {product &&
-                  product.availability &&
-                  product.availability.available !== undefined ? (
+                  product.pricing &&
+                  product.pricing.available !== undefined ? (
```
Comment (dominik-zeglen) [inline saleor/static/dashboard-next/components/ProductList/ProductList.tsx]: We can use `maybe(() => product.pricing.available) !== undefined` here.

File: saleor/static/dashboard-next/components/ProductList/ProductList.tsx [hunk 6]
```diff
@@ -174,10 +174,10 @@ export const ProductList = withStyles(styles, { name: "ProductList" })(
                 </TableCell>
                 <TableCell className={classes.colPrice}>
                   {product &&
-                  product.price &&
-                  product.price.amount !== undefined &&
-                  product.price.currency !== undefined ? (
-                    <Money money={product.price} />
+                  product.basePrice &&
+                  product.basePrice.amount !== undefined &&
+                  product.basePrice.currency !== undefined ? (
```
Comment (dominik-zeglen) [inline saleor/static/dashboard-next/components/ProductList/ProductList.tsx]: Same

File: saleor/static/dashboard-next/orders/components/OrderFulfillmentDialog/OrderFulfillmentDialog.tsx [hunk 7]
```diff
@@ -121,7 +121,7 @@ const OrderFulfillmentDialog = withStyles(styles, {
                       product.quantity - product.quantityFulfilled;
                     return (
                       <TableRow key={product.id}>
-                        <TableCellAvatar thumbnail={product.thumbnailUrl} />
+                        <TableCellAvatar thumbnail={product.thumbnail.url} />
```
Comment (dominik-zeglen) [inline saleor/static/dashboard-next/orders/components/OrderFulfillmentDialog/OrderFulfillmentDialog.tsx]: `maybe(() => product.thumbnail.url)`

File: saleor/static/dashboard-next/orders/components/OrderUnfulfilledItems/OrderUnfulfilledItems.tsx [hunk 8]
```diff
@@ -75,7 +75,7 @@ const OrderUnfulfilledItems = withStyles(styles, {
             hover={!!line}
             key={maybe(() => line.id)}
           >
-            <TableCellAvatar thumbnail={line.thumbnailUrl} />
+            <TableCellAvatar thumbnail={line.thumbnail.url} />
```
Comment (dominik-zeglen) [inline saleor/static/dashboard-next/orders/components/OrderUnfulfilledItems/OrderUnfulfilledItems.tsx]: `maybe(() => line.thumbnail.url)`

File: saleor/static/dashboard-next/orders/containers/OrderOperations.tsx [hunk 9]
```diff
@@ -71,75 +71,73 @@ import { OrderVoid, OrderVoidVariables } from "../types/OrderVoid";
...
+    >;
+    orderLinesAdd: PartialMutationProviderOutput<
+      OrderLinesAdd,
+      OrderLinesAddVariables
+    >;
+    orderLineUpdate: PartialMutationProviderOutput<
+      OrderLineUpdate,
+      OrderLineUpdateVariables
+    >;
+  }) => React.ReactNode;
```
Comment (dominik-zeglen) [inline saleor/static/dashboard-next/orders/containers/OrderOperations.tsx]: Prettier?

Comment (benekex2) [reply saleor/static/dashboard-next/orders/containers/OrderOperations.tsx]: I think so

File: saleor/static/dashboard-next/categories/components/CategoryProducts/CategoryProducts.tsx [hunk 10]
```diff
@@ -95,7 +97,9 @@ export const ProductList = withStyles(styles, { name: "ProductList" })(
             products,
             product => (
               <TableRow key={product ? product.id : "skeleton"}>
-                <TableCellAvatar thumbnail={product && product.thumbnailUrl} />
+                <TableCellAvatar
+                  thumbnail={product && maybe(() => product.thumbnail.url)}
```
Comment (dominik-zeglen) [inline saleor/static/dashboard-next/categories/components/CategoryProducts/CategoryProducts.tsx]: No need to put `product &&` here, maybe handles full path

File: saleor/static/dashboard-next/components/ProductList/ProductList.tsx [hunk 11]
```diff
@@ -152,31 +152,27 @@ export const ProductList = withStyles(styles, { name: "ProductList" })(
...
                 <TableCell className={classes.colPrice}>
                   {product &&
-                  product.price &&
-                  product.price.amount !== undefined &&
-                  product.price.currency !== undefined ? (
-                    <Money money={product.price} />
+                  maybe(() => product.basePrice) &&
+                  maybe(() => product.basePrice.amount) !== undefined &&
+                  maybe(() => product.basePrice.currency) !== undefined ? (
+                    <Money money={product.basePrice} />
```
Comment (dominik-zeglen) [inline saleor/static/dashboard-next/components/ProductList/ProductList.tsx]: Check if this one works

```suggestion
                    <TableCell className={classes.colPrice}>
                      {maybe(
                        () => product.price.amount && product.price.currency
                      ) ? (
                        <Money money={product.price} />
                      ) : (
                         <Skeleton />
                      )}
                     </TableCell>
```

Comment (benekex2) [reply saleor/static/dashboard-next/components/ProductList/ProductList.tsx]: Nope, it does not show zero values

**Pull request merged**